### PR TITLE
feat(common): add a temporary failsafe to filter dragonfish and cobia SCALE users that forgot to follow instructions

### PR DESCRIPTION
**Pull request by Ornias1993**
**Description**
This is a **TEMPORARY** hotfix to ensure SCALE users that forgot setting their `storageClass` to `SCALE-ZFS`, loose data on update to dragonfish.

This will be removed in the future, likely together with support for SCALE storageClasses all-together.

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

**📃 Notes:**

**✔️ Checklist:**

- [ ] ⚖️ My code follows the style guidelines of this project
- [ ] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

### CODE REVIEW

File: library/common/templates/lib/storage/_storageClassName.tpl [hunk 1]
```diff
@@ -37,11 +37,20 @@ objectData: The object data of the pvc
       {{- $className = tpl $storageClass $rootCtx -}}
     {{- end -}}
 
-  {{- else if $rootCtx.Values.global.ixChartContext -}}
-    {{- if not $rootCtx.Values.global.ixChartContext.storageClassName -}}
-      {{- fail (printf "%s - Expected non-empty [global.ixChartContext.storageClassName]" $caller) -}}
+  {{- else if and $rootCtx.Values.global.ixChartContext $rootCtx.Values.global.ixChartContext.storageClassName -}}
+    {{- $scaleClassFound := false -}}
+    {{- with (lookup "v1" "StorageClass" "" $rootCtx.Values.global.ixChartContext.storageClassName -}}
```
Comment (Ornias1993) [inline library/common/templates/lib/storage/_storageClassName.tpl:42]:
```suggestion
    {{- with (lookup "v1" "StorageClass" "" $rootCtx.Values.global.ixChartContext.storageClassName ) -}}
```

File: library/common/values.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ global:
   fallbackDefaults:
     # -- Define a storageClassName that will be used for all PVCs
     # Can be overruled per PVC
-    storageClass:
+    storageClass: ""
```
Comment (Ornias1993) [inline library/common/values.yaml:13]:
```suggestion
    storageClass:
```

File: library/common/templates/lib/storage/_storageClassName.tpl [hunk 3]
```diff
@@ -37,11 +37,20 @@ objectData: The object data of the pvc
...
+    {{- with (lookup "v1" "StorageClass" "" $rootCtx.Values.global.ixChartContext.storageClassName ) -}}
+      {{/* Check if there is an actually valid storageClass found */}}
+      {{- if .provisioner -}}
+        {{- $scaleClassFound = true -}}
+      {{- end -}}
+    {{- end -}}
+
+    {{- if $scaleClassFound -}}
+      {{- $className = tpl $rootCtx.Values.global.ixChartContext.storageClassName $rootCtx -}}
+    {{- else -}}
```
Comment (Ornias1993) [inline library/common/templates/lib/storage/_storageClassName.tpl:51]:
```suggestion
    {{- else if $rootCtx.Values.global.fallbackDefaults.storageClass -}}
```

**Comment by Ornias1993** [comment]
[DNM] This will inherently break compatibility with Cobia!!!

File: library/common/Chart.yaml [hunk 4]
```diff
@@ -15,7 +15,7 @@ maintainers:
 name: common
 sources: null
 type: library
-version: 20.1.2
+version: 20.2.0
```
Comment (Ornias1993) [inline library/common/Chart.yaml:18]:
```suggestion
version: 21.0.0
```

File: library/common/Chart.yaml [hunk 5]
```diff
@@ -15,7 +15,7 @@ maintainers:
 name: common
 sources: null
 type: library
-version: 20.1.2
+version: 21.0.0
```
Comment (Ornias1993) [inline library/common/Chart.yaml:18]:
```suggestion
version: 20.2.0
```

**Review by stavros-k — APPROVED** [review]

**Pull request merged**